packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1475,21 +1200,354 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
+            const SizedBox(height: 2.0),
+        ],
+      ),
+    );
+  }
+}
+
+class _HourMinuteTextField extends StatefulWidget {
+  const _HourMinuteTextField({
+    Key key,
+    @required this.selectedTime,
+    @required this.isHour,
+    @required this.style,
+    @required this.validator,
+    @required this.onSavedSubmitted,
+    this.onChanged,
+  }) : super(key: key);
+
+  final TimeOfDay selectedTime;
+  final bool isHour;
+  final TextStyle style;
+  final FormFieldValidator<String> validator;
+  final ValueChanged<String> onSavedSubmitted;
+  final ValueChanged<String> onChanged;
+
+  @override
+  _HourMinuteTextFieldState createState() => _HourMinuteTextFieldState();
+}
+
+class _HourMinuteTextFieldState extends State<_HourMinuteTextField> {
+  TextEditingController controller;
+  FocusNode focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    focusNode = FocusNode()..addListener(() {
+      setState(() { }); // Rebuild.
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    controller ??= TextEditingController(text: _formattedValue);
+  }
+
+  String get _formattedValue {
+    final bool alwaysUse24HourFormat = MediaQuery.of(context).alwaysUse24HourFormat;
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    return widget.isHour ? localizations.formatHour(
+      widget.selectedTime,
+      alwaysUse24HourFormat: alwaysUse24HourFormat,
+    ) : localizations.formatMinute(widget.selectedTime);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    final InputDecorationTheme inputDecorationTheme = TimePickerTheme.of(context).inputDecorationTheme;
+    InputDecoration inputDecoration;
+    if (inputDecorationTheme == null) {
+      inputDecoration = InputDecoration(
+        contentPadding: const EdgeInsetsDirectional.only(bottom: 16.0, start: 3.0),
+        filled: true,
+        fillColor: focusNode.hasFocus ? colorScheme.surface : colorScheme.onSurface.withOpacity(0.12),
+        enabledBorder: const OutlineInputBorder(
+          borderSide: BorderSide(color: Colors.transparent),
+        ),
+        errorBorder: OutlineInputBorder(
+          borderSide: BorderSide(color: colorScheme.error, width: 2.0),
+        ),
+        focusedBorder: OutlineInputBorder(
+          borderSide: BorderSide(color: colorScheme.primary, width: 2.0),
+        ),
+        focusedErrorBorder: OutlineInputBorder(
+          borderSide: BorderSide(color: colorScheme.error, width: 2.0),
+        ),
+        hintStyle: widget.style.copyWith(color: colorScheme.onSurface.withOpacity(0.36)),
+        // TODO(rami-a): Remove this logic once https://github.com/flutter/flutter/issues/54104 is fixed.
+        errorStyle: const TextStyle(fontSize: 0.0, height: 0.0), // Prevent the error text from appearing.
+      );
+    } else {
```

Comment:
Sometimes this kind of thing is easier to read if the short and simple clause is first

Reply:
Done